From 721932457c96ad376903d74d171d2d5986096899 Mon Sep 17 00:00:00 2001
From: Dmitry Kasatkin <d.kasatkin@samsung.com>
Date: Sun, 26 Oct 2014 12:42:07 +0200
Subject: [PATCH 09/45] integrity: forbid write access when reading data to the
 buffer

In general when kernel is reading a file it does not want anyone writing to it.
This patch denies write access before reading the file content.

Signed-off-by: Dmitry Kasatkin <d.kasatkin@samsung.com>
---
 security/integrity/iint.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/security/integrity/iint.c b/security/integrity/iint.c
index 1fd7fc9..9fef48b 100644
--- a/security/integrity/iint.c
+++ b/security/integrity/iint.c
@@ -222,6 +222,10 @@ int __init integrity_read_file(const char *path, char **data)
 		return rc;
 	}
 
+	rc = deny_write_access(file);
+	if (rc)
+		goto out_fput;
+
 	size = i_size_read(file_inode(file));
 	if (size <= 0)
 		goto out;
@@ -240,6 +244,8 @@ int __init integrity_read_file(const char *path, char **data)
 	else
 		*data = buf;
 out:
+	allow_write_access(file);
+out_fput:
 	fput(file);
 	return rc;
 }
-- 
2.1.4

